spinalcordtoolbox/scripts/sct_compute_compression.py

```diff
@@ -0,0 +1,650 @@
+
+
+NEAR_ZERO_THRESHOLD = 1e-6
+
+
+# PARSER
+# ==========================================================================================
+def get_parser():
+    parser = SCTArgumentParser(
+        description='Compute normalized morphometric measures to assess spinal cord compression.\n'
+                    '\n'
+                    'Metrics are normalized using the non-compressed levels above and below the compression site using '
+                    'the following equation:\n'
+                    '\n'
+                    '    ratio = (1 - mi/((ma+mb)/2))\n'
+                    '\n'
+                    'Where mi: metric at the compression level, ma: metric above the compression level, mb: metric '
+                    'below the compression level.\n'
+                    '\n'
+                    'Additionally, if the "-normalize" flag is used, metrics are normalized using a database built '
+                    'from healthy control subjects. This database uses the PAM50 template as an anatomical reference '
+                    'system.\n'
+                    '\n'
+                    'Reference: Miyanji F, Furlan JC, Aarabi B, Arnold PM, Fehlings MG. Acute cervical traumatic '
+                    'spinal cord injury: MR imaging findings correlated with neurologic outcome--prospective study '
+                    'with 100 consecutive patients. Radiology 2007;243(3):820-827.'
+    )
+
+    mandatory = parser.add_argument_group("\nMANDATORY ARGUMENTS")
+    mandatory.add_argument(
+        '-i',
+        metavar=Metavar.file,
+        required=True,
+        help='Spinal cord segmentation mask to compute morphometrics. Example: sub-001_T2w_seg.nii.gz'
+             '\nNote: If no normalization is wanted (i.e., if the "-normalize" flag is not specified),'
+             ' metric ratio will take the average along the segmentation centerline.'
+    )
+    mandatory.add_argument(
+        '-vertfile',
+        metavar=Metavar.file,
+        required=True,
+        help='Vertebral labeling file. Example: sub-001_T2w_seg_labeled.nii.gz'
+             'Note: The input and the vertebral labelling file must be in the same voxel coordinate system'
+             'and must match the dimensions between each other.'
+    )
+    mandatory.add_argument(
+        '-l',
+        metavar=Metavar.file,
+        required=True,
+        help='NIfTI file that includes labels at the compression sites. '
+             'Each compression site is denoted by a single voxel of value `1`.'
+             ' Example: sub-001_T2w_compression_labels.nii.gz '
+             'Note: The input and the compression label file must be in the same voxel coordinate system '
+             'and must match the dimensions between each other.'
+    )
+    mandatory.add_argument(
+        '-metric',
+        required=False,
+        help='Metric to normalize. ',
+        default='diameter_AP',
+        choices=['diameter_AP', 'area', 'diameter_RL', 'eccentricity', 'solidity'],
```

Comment:
I notice that in the `compute_shape()` function in `process_seg`, there is a similar list of metrics:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/a029f36f7bae57cdf238ba13bc416f5edbc8dcbc/spinalcordtoolbox/process_seg.py#L33-L43

And also in the `save_csv()` function in `aggregate_slicewise.py`: 

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/a029f36f7bae57cdf238ba13bc416f5edbc8dcbc/spinalcordtoolbox/aggregate_slicewise.py#L540-L546

Some naive questions here:

- Why are some metrics included, but others excluded?
- Is there any way to reuse a single list of metrics? e.g. by making it a module-level constant in `process_seg.py`, then importing it here? (Just to make sure the available metrics don't become out of sync.)

Reply:
Good point! 
The thing is that it makes sense to do the normalization only for some metrics (`choices=['diameter_AP', 'area', 'diameter_RL', 'eccentricity', 'solidity']`).